### PR TITLE
GH-425 Setting the right default for access_token attribute

### PR DESCRIPTION
**Pull request by maheshjfrog**
Fixes: #425

### CODE REVIEW

**Review by alexhung — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## 6.5.1 (Apr 20, 2022). Tested on Artifactory 7.37.14
+
+BUG FIXES:
+
+* provider:  Setting the right default value for 'access_token' attribute. [GH-426]
+
```
Comment (chb0github) [inline CHANGELOG.md:6]: Please make sure we can directly link to the issue from the change log

**Review by chb0github — DISMISSED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
...
 ## 6.5.0 (Apr 15, 2022). Tested on Artifactory 7.37.14
```
Comment (tjstansell) [inline CHANGELOG.md:7]: hm... this date for 6.5.0 seems wrong too, wanna fix that too while you're here?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,10 @@
-## 6.5.0 (Apr 15, 2022). Tested on Artifactory 7.37.14
+## 6.5.1 (Apr 20, 2022). Tested on Artifactory 7.37.14
+
+BUG FIXES:
+
+* provider:  Setting the right default value for 'access_token' attribute. PR: [GH-426](https://github.com/jfrog/terraform-provider-artifactory/pull/426). Issue [#425](https://github.com/jfrog/terraform-provider-artifactory/issues/425)
```
Comment (alexhung) [inline CHANGELOG.md:5]: No need to link to the issue. Just the PR.

**Review by alexhung — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (maheshjfrog) [reply CHANGELOG.md:5]: @chb0github requested for link to issue so added.

Comment (chb0github) [reply CHANGELOG.md:5]: You can do a relative link - you don't need the path. @alexhung I just want that so when you look at the changelog the ticket number is clickable

File: pkg/artifactory/provider.go [hunk 4]
```diff
@@ -126,8 +126,8 @@ func Provider() *schema.Provider {
 				Type:        schema.TypeString,
 				Optional:    true,
 				Sensitive:   true,
-				DefaultFunc: schema.MultiEnvDefaultFunc([]string{"ARTIFACTORY_ACCESS_TOKEN", "JFROG_ACCESS_TOKEN"}, ""),
-				Description: "This is a access token that can be given to you by your admin under `Identity and Access`",
+				DefaultFunc: schema.MultiEnvDefaultFunc([]string{"ARTIFACTORY_ACCESS_TOKEN", "JFROG_ACCESS_TOKEN"}, nil),
+				Description: "(Optional) This is a access token that can be given to you by your admin under `Identity and Access`. Default to 'nil'.",
```
Comment (chb0github) [inline pkg/artifactory/provider.go:130]: `nil` means nothing to the end user. You have to explain the behavior they can expect. `If not set, the API_KEY will be used` For example.

**Review by chb0github — CHANGES_REQUESTED** [review]
see above

File: CHANGELOG.md (hunk 3 above)
Comment (alexhung) [reply CHANGELOG.md:5]: Sure, I'm all for PR to be linked. Just no need to link to issue as well.

Comment (maheshjfrog) [reply CHANGELOG.md:5]: Ok ... Leaving as is. 
ie: both PR and Issue mentioned.

**Review by alexhung — APPROVED** [review]

**Review by chb0github — APPROVED** [review]

**Pull request merged**